### DISCUSSION TITLE
Can I Link internal and external tables as part of a relational DB?

### DISCUSSION THREAD

**Discussion by KernOil** [Budibase/budibase#2828] [Ideas]
I am using MYSQL for my relational DB I would like to have the ability to save attachments in my BB app however to the best of my knowledge this is not an available feature yet via the SQL data connector.

Is there a way for me to use the internal Couch DB for the attachments and have it relate to the SQL tables I pull in?

**Comment by mjashanks** [comment]
I've spent a while trying to work out if this is possible. The answer is no, for the moment.

This is something that I can really see the utility of, and something that we have talked about before. So, I'm turning this into an "Idea" because I think it would be useful.

The issues that I ran into while trying this out.
1. It's hard using Multiple, nested data providers. Its difficult to understand the series of Data providers + Repeaters required.
2. I was trying to load 2 records into the same detail screen, then save both records with one click. This required me to also nest forms.

To achieve this sort of thing, I feel like we would have to
1. Reduce nesting of data providers. Could a data provider change, to allow it to fetch multiple sets of records, and link them?
2. Similar idea for forms. Should it be able to handle multiple concurrent schemas? I don't know how this would work, to be honest.

**Comment by KernOil** [comment]
@mjashanks Thank you for looking into this for me I really appreciate it! I look forward to this functionality in the future.

**Reply by dncpax** [reply]
@mjashanks I've seen this done whith a redis backend, where results from other datasources are saved in redis. Then a new query is made to redis, that joins the 2 other queries results by using a common column.
You would have to manage refreshing the queries, being one option to re-retrieve them when the redis query is run.

Hope I was clear enough...

**Comment by TakitoLoko** [comment]
It looks like it is possible now: https://budibase.com/blog/data/how-to-integrate-multiple-databases/

**Comment by sergey-vin** [comment]
Hi @mjashanks and other Budibase guys.
I think i've been able to achieve the 2-table binding but it was far from perfect.
I had 2 nested sets of data providers, repeaters and forms to achieve that, and it kind of worked. 
The schema was as following:

```
Data Provider 1
    +-> Repeater 1
        +-> Form 1

            +-> Data Provider 2
                +-> Repeater 2
                    +-> Form 2

                        +-> Title container
                            +-> delete button     <--- Those buttons have to be here 
                            +-> save button        <--- to have access to save both rows

                        +-> Field group 2
                            +-> ...fields...
                            +-> ...fields...

            +-> Field group 1
                +-> ... fields...
                +-> ... fields...
```

This was ugly, because it required me to have delete & save buttons inside 2 forms, because otherwise they won't have access to validate forms and save rows.
I'm trying to fix it now by using some sort of automation, internal "state" or something else, but I have a sense that this will be a sub-optimal solution - since only when I'm inside a form I'll be able to have validations. If I move outside of a form, I'll lose it. Let's see.

But yes, if there's a better way of handling the actions - like if I can have access to forms outside of my scope - that would solve everything. 
I hope the Budibase team can take that into account.

Obviously the other alternative - is to add attachments to mysql/pgsql. But being able to save to 2 tables is a bigger scope.

PS. As for https://budibase.com/blog/data/how-to-integrate-multiple-databases/ - it didn't look like a proper solution, since it managed consistency and validations quite poorly. We need a better way to have 2+ tables save support. Then right here right now we'd see if there's a problem with validation, and not hide it to some hidden automated jobs or replication jobs.
